Doc/library/textwrap.rst

```diff
@@ -17,31 +17,48 @@ If you're just wrapping or filling one or two text strings, the convenience
 functions should be good enough; otherwise, you should use an instance of
 :class:`TextWrapper` for efficiency.
 
-.. function:: wrap(text, width=70, **kwargs)
+.. function:: wrap(text, width=70, *, initial_indent="", \
+                   subsequent_indent="", expand_tabs=True, \
+                   replace_whitespace=True, fix_sentence_endings=False, \
+                   break_long_words=True, drop_whitespace=True, \
+                   break_on_hyphens=True, tabsize=8, *, max_lines=None, \
+                   **kwargs)
 
    Wraps the single paragraph in *text* (a string) so every line is at most
    *width* characters long.  Returns a list of output lines, without final
-   newlines.
+   newlines.  *width* defaults to ``70``.
 
-   Optional keyword arguments correspond to the instance attributes of
-   :class:`TextWrapper`, documented below.  *width* defaults to ``70``.
+   See :class:`TextWrapper` for documentation on all optional keyword
+   arguments: :attr:`.initial_indent`, :attr:`.subsequent_indent`,
+   :attr:`.expand_tabs`, :attr:`.replace_whitespace`,
+   :attr:`.fix_sentence_endings`, :attr:`.break_long_words`,
+   :attr:`.drop_whitespace`, :attr:`.break_on_hyphens` :attr:`.tabsize`, and
+   :attr:`.max_lines`.  Be careful not to misspell keyword arguments.
+   Misspelled keyword arguments will silently have no effect, and an exception
+   will not be raised.
```

Comment:
In my view repetition in the docs is not better than repetition in code.

Reply:
The root cause of the repetition is that these functions all recieve a laundry list of keyword arguments, and they are all the same or highly similar. In that sense, the API is repetitive. Maybe it could have been designed better if we had dataclasses for grouping the values passed in or something like that, but I think the repetition in the documentation is a  consequence of the library'a design, and we aren't going to change the API design.

Reply:
In the code, this list of args appears once in the constructor of  TextWrapper. The module-level wrapper functions just forward their kwargs to this constructor. It would be enough to define all these args once and in other places say "the meaning of the parameters are the same as over there".

It's easy for these repetitions to go out of sync with each other (if you add an arg to TextWrapper.__init__ you need to update the doc in multiple places).

I don't see what the benefit of the repetition is. For a user trying to study the library for the first time it's just tedious. For someone looking up a single function, clicking on a link to the TextWrapper constructor is not complicated.

Reply:
My preference would also be to avoid repeated list of links. I previously suggested inline links as a compromise between no links and the "tall" list with a link per line as in initial revision. I don't have a strong opinion though, as long as it's not a "tall" list.

Reply:
@iritkatriel I'm sorry, I thought you were talking about repetition between the function signature and the inline links but now I realize I don't think that's what you're talking about. I'm going to take a closer look later today, sorry for any confusion.

Reply:
Ah, sorry I was vague.

Reply:
@iritkatriel I see what you are saying. I went ahead and removed the repetitive links all together. Do you think that the explicit function signatures is a good change? Otherwise, there's nothing to be done here.

Reply:
Good question, I wonder if the documentation working group can come up with suggestions. This is not the only library that has this pattern. @JulienPalard @Mariatta